pip/req/req_install.py

```diff
@@ -66,6 +68,23 @@ def _strip_extras(path):
     return path_no_extras, extras
 
 
+class Requirement(BaseRequirement):
+    '''
+    Retro compat derived class that assure the access to a project_name
+    attribute, to be retro-compatible with code that relied on
+    pkg_resources.Requirement that pip "Requirement.parse_requirements"
+    would have returned in releases previous to 8.1.2
+    '''
+
+    @property
+    def project_name(self):
+        return self.name
```

Comment:
In the original class the project name was being processed via a regex, I just copied the code when I updated salt but it might make sense to do the same here